store/sqlstore/post_store.go

```diff
@@ -822,7 +827,7 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
 						Id = ChannelId
 							AND (TeamId = :TeamId OR TeamId = '')
 							AND UserId = :UserId
-							AND DeleteAt = 0
+							` + deletedQueryPart + `
```

Comment:
Please can you add a couple of extra unit tests to make sure that searching posts in deleted channels works.

Reply:
Sure